src/components/screens/DashboardGestaoProduto/index.jsx

```diff
@@ -42,59 +56,159 @@ export default class DashboardGestaoProduto extends Component {
+        tituloToBeFiltered.length >= 3 &&
+        (data.titulo_produto = tituloToBeFiltered);
+      this.setState({ loading: true });
+      const response = await getHomologacoesPorTituloMarca(data);
+      if (response.status === HTTP_STATUS.OK) {
+        const cards = listarCardsPermitidos();
+        const ehTerceirizada = this.ehPerfilTerceirizada();
+        cards.forEach(card => {
+          card.items = incluirDados(card.incluir_status, response.data.results);
+        });
+        this.setState({
+          cards: cards,
+          cardsFiltered: cards.concat(),
+          ehTerceirizada,
+          loading: false
+        });
+        resolve();
+      } else {
+        this.setState({ loading: false });
+        reject(response.errors);
+      }
     });
   };
 
-  filtrarMarca(listaFiltro, value) {
-    const wordToFilter = slugify(value.toLowerCase());
-    return listaFiltro.map(card => {
-      card.items = card.items.filter(
-        item =>
-          slugify(item.marca_produto.toLowerCase()).search(wordToFilter) !==
-            -1 ||
-          slugify(item.id_externo.toLowerCase()).search(wordToFilter) !== -1
-      );
-      return card;
-    });
+  filtrarTitulo = () => {
+    const {
+      filtroPorTituloAtivo,
+      tituloToBeFiltered,
+      marcaToBeFiltered
+    } = this.state;
+    if (tituloToBeFiltered.length >= 3) {
+      try {
+        this.filtraHomologacoesPorTituloMarca();
+        this.setState({
+          filtroPorTituloAtivo: true
+        });
+      } catch (e) {
+        this.setState({ erro: true });
+      }
+    } else if (
+      tituloToBeFiltered.length < 3 &&
+      marcaToBeFiltered.length >= 3 &&
+      filtroPorTituloAtivo
+    ) {
+      this.filtraHomologacoesPorTituloMarca();
+      this.setState({
+        filtroPorTituloAtivo: false,
+        filtroPorMarcaAtivo: true
+      });
+    } else {
+      if (this.state.filtroPorTituloAtivo) {
+        this.loadDashBoardGestaoProdutos();
+        this.setState({
+          filtroPorTituloAtivo: false
+        });
+      }
+    }
+  };
+
+  filtrarMarca() {
+    const {
+      marcaToBeFiltered,
+      tituloToBeFiltered,
+      filtroPorMarcaAtivo
+    } = this.state;
+    if (marcaToBeFiltered.length >= 3) {
+      try {
+        this.filtraHomologacoesPorTituloMarca();
+        this.setState({
+          filtroPorMarcaAtivo: true
+        });
+      } catch (e) {
+        this.setState({ erro: true });
+      }
+    } else if (
+      marcaToBeFiltered.length < 3 &&
+      tituloToBeFiltered.length >= 3 &&
+      filtroPorMarcaAtivo
+    ) {
+      this.filtraHomologacoesPorTituloMarca();
+      this.setState({
+        filtroPorMarcaAtivo: false,
+        filtroPorTituloAtivo: true
+      });
+    } else {
+      if (this.state.filtroPorMarcaAtivo) {
+        this.loadDashBoardGestaoProdutos();
+        this.setState({
+          filtroPorMarcaAtivo: false
+        });
+      }
+    }
   }
 
+  onPesquisaChanged = values => {
```

Comment:
clearTimeout(this.typingTimeout);
    this.typingTimeout = setTimeout(() => {
data = { };
if (values.titulo && values.titulo.length > 2) {
    data['titulo'] = values.titulo;
}
if (values.marca && values.marca.length > 2) {
  data['marca'] = values.marca;
}
this.setState({ loading: true });
      const response = await getHomologacoesPorTituloMarca(data);
      if (response.status === HTTP_STATUS.OK) {
        const cards = listarCardsPermitidos();
        const ehTerceirizada = this.ehPerfilTerceirizada();
        cards.forEach(card => {
          card.items = incluirDados(card.incluir_status, response.data.results);
        });
        this.setState({
          cards: cards,
          cardsFiltered: cards.concat(),
          ehTerceirizada,
          loading: false
        });
        resolve();
      } else {
        this.setState({ loading: false });
        reject(response.errors);
      }
}, 1000);

Reply:
alterações realizadas